ethpm_types/contract_type.py

```diff
@@ -478,3 +498,17 @@ def _get_first_instance(self, _type: Type[ABI_SINGLETON_T]) -> Optional[ABI_SING
             return abi
 
         return None
+
+    @cached_property
+    def _abi_identifiers(self) -> List[Tuple[ABI_W_SELECTOR_T, str]]:
+        def get_id(aitem: ABI) -> str:
+            if isinstance(aitem, MethodABI) or isinstance(aitem, ErrorABI):
+                return HexBytes(self._selector_hash_fn(aitem.selector)[:4]).hex()
+            else:
+                assert hasattr(aitem, "selector")
```

Comment:
i really don like asserts in production code. Is his needed?
They always fail with little context and dont work outside of debug mode anyway.

why not just:

```python
elif selector := getattr(aitem, "selector", None):
    return HexBytes(self._selector_hash_fn(selector)).hex()
```

Reply:
Sorry, I thought this was common use in Ape.  Will avoid in the future.

Your suggestion doesn't really mean the same thing though.  That leaves a (probably) impossible logic branch where the func doesn't return what we expect.  I think I have a better fix for this though so I'll put together a quick PR for this since automerge did its thing.

Reply:
the only time i use asserts in prod is for type-checkers where i know 100% the error will never actually raise but lets type checkers know the type, maybe that is the case here! in which case, a comment suffices in pacifying my OCD.